### PR TITLE
Aw/core 107 validate

### PR DESCRIPTION
**Pull request by zredlined**
Adding model validation as a default option to detect and prevent model overfitting. Introduces new boolean config parameter `validation_split`. If true, split 20% of the database to use for validation, and default to using `validation_loss` as the best metric for model evaluation. If false, do not use validation dataset, and default to `loss` as the best metric.

Example output with validation_split=True

Standard output
```
Epoch 1/2
4/4 [==============================] - 5s 874ms/step - loss: 4.4493 - accuracy: 0.0514 - val_loss: 3.6920 - val_accuracy: 0.1138
Epoch 2/2
4/4 [==============================] - 2s 557ms/step - loss: 3.7196 - accuracy: 0.0811 - val_loss: 3.3155 - val_accuracy: 0.0528
```

model_history.csv
```
epoch,loss,accuracy,val_loss,val_accuracy,best
0,4.922,0.0646,3.692,0.1138,0
1,3.6756,0.0815,3.3155,0.0528,1
```

with validation_split=False
```
6/6 [==============================] - 4s 395ms/step - loss: 4.2533 - accuracy: 0.0547
Epoch 2/2
6/6 [==============================] - 2s 404ms/step - loss: 3.3692 - accuracy: 0.0782
```

model_history.csv
```
epoch,loss,accuracy,val_loss,val_accuracy,best
0,4.3805,0.0615,0,0,0
1,3.2814,0.0871,0,0,1
```

### CODE REVIEW

**Comment by johntmyers** [comment]
I think the dataclass for the `EpochState` needs to be updated with the new params that are being captured by custom callbacks:

https://github.com/gretelai/gretel-synthetics/blob/3570b09cb17d455b9f776e0acbb001eae8244f0f/src/gretel_synthetics/train.py#L36

**Review by johntmyers — APPROVED** [review]
